pkg/controllers/job/job_controller_actions.go

```diff
@@ -106,14 +107,10 @@ func (cc *Controller) killJob(jobInfo *apis.JobInfo, nextState state.NextStateFn
 	}
 
 	// Update Job status
-	if job, err := cc.vkClients.BatchV1alpha1().Jobs(job.Namespace).UpdateStatus(job); err != nil {
+	if err := cc.updateJobStatus(job); err != nil {
 		glog.Errorf("Failed to update status of Job %v/%v: %v",
 			job.Namespace, job.Name, err)
 		return err
-	} else {
-		if e := cc.cache.Update(job); e != nil {
```

Comment:
We donot need update at all, now every time we `processNextReq`, get from the informer lister and do a deepcopy to prevent concurrent status modification lead to unexpected result.